psi4/src/psi4/libmints/vector.h

```diff
@@ -204,7 +204,7 @@ class Vector
     int dim(int h = 0) const { return dimpi_[h]; }
 
     /// Returns the dimension array
-    int *dimpi() const { return dimpi_; }
+    int *dimpi() { return dimpi_; }
```

Comment:
It might be worth switching this over to return a `const Dimension` object instead of the implicit casting here so that the functionality is similar to the Matrix class.